services/src/main/java/org/keycloak/authentication/DefaultAuthenticationFlow.java

```diff
@@ -153,10 +154,42 @@ public Response processAction(String actionExecution) {
         Response response = processResult(result, true);
         if (response == null) {
             processor.getAuthenticationSession().removeAuthNote(AuthenticationProcessor.CURRENT_AUTHENTICATION_EXECUTION);
+            checkAndValidateParentFlow(model);
             return processFlow();
         } else return response;
     }
 
+    /**
+     * Removes the execution status for an execution. If it is a flow, do the same for all sub-executions.
+     * @param execution the execution for which the status must be cleared
+     */
+    private void recursiveClearExecutionStatus(AuthenticationExecutionModel execution) {
+        processor.getAuthenticationSession().getExecutionStatus().remove(execution.getId());
+        if (execution.isAuthenticatorFlow()) {
+            List<AuthenticationExecutionModel> flowExecutions = processor.getRealm().getAuthenticationExecutions(execution.getFlowId());
```

Comment:
Can be shorten:
processor.getRealm().getAuthenticationExecutions(execution.getFlowId()).forEach(this::recursiveClearExecutionStatus);